src/Aquarius.ONE.ClientSDK/Common/Configuration/ConfigurationApi.cs

```diff
@@ -314,5 +314,193 @@ public async Task<bool> DeleteConfigurationAsync(string id)
+                return null;
+            }
+            catch (Exception e)
+            {
+                Event(e, new ClientApiLoggerEventArgs { EventLevel = EnumEventLevel.Error, Module = "ConfigurationApi", Message = $"GetConfigurationNotesAsync Failed - {e.Message}" });
+                throw;
+            }
+        }
+
+        public async Task<List<proto.ConfigurationTag>> GetConfigurationTagsAsync(string configurationId)
+        {
+            var watch = System.Diagnostics.Stopwatch.StartNew();
+
+            var requestId = Guid.NewGuid();
+
+            List<proto.ConfigurationTag> configurationTags = new List<proto.ConfigurationTag>();
+
+            string endpointUrl = $"common/configuration/v2/notes/{configurationId}/uniquetags";
+
+            try
+            {
+                var respContent = await _restHelper.GetRestProtocolBufferAsync(requestId, endpointUrl).ConfigureAwait(_continueOnCapturedContext);
+                if (respContent.ResponseMessage.IsSuccessStatusCode)
+                {
+                    var results = respContent.ApiResponse.Content.ConfigurationTags.Items.Distinct().ToList();
+                    foreach (var result in results)
+                    {
+                        ConfigurationTag configurationTag = new proto.ConfigurationTag(result);
+                        configurationTags.Add(configurationTag);
+                    }
+                    Event(null, new ClientApiLoggerEventArgs { EventLevel = EnumEventLevel.Trace, HttpStatusCode = respContent.ResponseMessage.StatusCode, ElapsedMs = watch.ElapsedMilliseconds, Module = "ConfigurationApi", Message = $"GetConfigurationsAsync Success" });
+                    return configurationTags;
+                }
+                Event(null, new ClientApiLoggerEventArgs { EventLevel = EnumEventLevel.Warn, HttpStatusCode = respContent.ResponseMessage.StatusCode, ElapsedMs = watch.ElapsedMilliseconds, Module = "ConfigurationApi", Message = $"GetConfigurationsAsync Failed" });
+                return null;
+            }
+            catch (Exception e)
+            {
+                Event(e, new ClientApiLoggerEventArgs { EventLevel = EnumEventLevel.Error, Module = "ConfigurationApi", Message = $"GetConfigurationTagsAsync Failed - {e.Message}" });
+                throw;
+            }
+        }
+
+        public async Task<bool> CreateConfigurationNoteAsync(ConfigurationNote configurationNote)
+        {
+            var watch = System.Diagnostics.Stopwatch.StartNew();
+
+            var requestId = Guid.NewGuid();
+            var endpoint = $"common/configuration/v2/notes";
+
+            var json = JsonConvert.SerializeObject(configurationNote, new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore });
+
+            try
+            {
+                var respContent = await _restHelper.PostRestJSONAsync(requestId, json, endpoint).ConfigureAwait(_continueOnCapturedContext);

```

Comment:
Steve indicated that the SDK supported sending protobuf requests which would be good to use instead of JSON if we can.

Reply:
we don't have Post protobuf request methods implemented in RestHelper class. We have it for Get and I used it in all Get requests